diff -u -r1.6 lokkit
--- system-config-firewall-1.0.5/src/lokkit	20 Aug 2007 15:57:57 -0000	1.6
+++ system-config-firewall-1.0.5/src/lokkit	31 Aug 2007 15:46:48 -0000
@@ -45,8 +45,6 @@
 # parse /etc/sysconfig/system-config-firewall or
 # /etc/sysconfig/system-config-securitylevel
 old_config = read_sysconfig_config(old_config)
-if not old_config:
-    old_config = [ ]
 
 # parse selinux config
 selinux_args = fw_selinux.read() or [ ]
