diff -up system-config-firewall-1.0.5/src/fw_sysconfig.py.sysconfig system-config-firewall-1.0.5/src/fw_sysconfig.py
--- system-config-firewall-1.0.5/src/fw_sysconfig.py.sysconfig	2007-08-17 17:24:58.000000000 +0200
+++ system-config-firewall-1.0.5/src/fw_sysconfig.py	2007-09-05 18:39:06.000000000 +0200
@@ -31,7 +31,7 @@ def read_sysconfig_config(merge_config=N
     try:
         fd = open(filename, 'r')
     except:
-        return None
+        return merge_config
     argv = [ ]
     for line in fd.xreadlines():
         if not line:
